standard/statements.md

```diff
@@ -1112,33 +1112,48 @@ If the *foreach_statement* contains both or neither `ref` and `readonly`, the it
 
 The iteration variable corresponds to a local variable with a scope that extends over the embedded statement. During execution of a `foreach` statement, the iteration variable represents the collection element for which an iteration is currently being performed. If the iteration variable denotes a read-only variable, a compile-time error occurs if the embedded statement attempts to modify it (via assignment or the `++` and `--` operators) or pass it as a reference or output parameter.
 
-The compile-time processing of a `foreach` statement first determines the ***collection type***, ***enumerator type*** and ***iteration type*** of the expression. The processing for a `foreach` statement is detailed in [§13.9.5.2](statements.md#13952-synchronous-foreach) and the process for an `await foreach` is detailed in [§13.9.5.3](statements.md#13953-await-foreach).
+The compile-time processing of a `foreach` statement first determines the ***collection type*** (`C`), ***enumerator type*** (`E`) and ***iteration type*** (`T`, `ref T` or `ref readonly T`) of the expression.
+
+The determination is similar for the synchronous and asynchronous versions. Different interfaces with different methods and return types distinguish the synchronous and asynchronous versions. The general process proceeds as follows. Names within '«' and '»' are placeholders for the actual names for synchronous and asynchronous iterators. The types allowed for «GetEnumerator», «MoveNext», «IEnumerable»\<T>, «IEnumerator»\<T>, and any other distinctions are detailed in [§13.9.5.2](statements.md#13952-synchronous-foreach) for a synchronous `foreach` statement, and in [§13.9.5.3](statements.md#13953-await-foreach) for an asynchronous `foreach` statement.
+
+1. Determine whether the type `X` has an appropriate «GetEnumerator» method:
```

Comment:
In this context, it's not clear where the type `X` comes from.

Reply:
Good point, but I don't think it's `C`. The ***collection type*** (`C`) is determined on like 1126 by examining `X`.

Reply:
The original text (original line 1126) had `If the type `X` of *expression*`. It was probably removed by mistake? @BillWagner